Git/config.sublime-syntax

```diff
@@ -0,0 +1,206 @@
+  zero_to_255: 25[0-5]|2[0-4][0-9]|1\d\d|[1-9][0-9]|[0-9]
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: section
+
+  boolean:
+    # 0 and 1 are omitted due to mismatches (e.g. "log -1", "HEAD~1"). They will
+    # be matched as unquoted strings instead.
+    - match: \b(?:yes|true|on|off|no|false){{space_or_eol}}
+      scope: constant.language.gitconfig
+
+  color:
+    - include: color-named      # red, blue, green
+    - include: color-attribute  # bold, italic, underline
+    - include: color-numeric    # 0-255
+
+  color-attribute:
+    - match: \b(?:no-?)?(?:ul|strike|reverse|italic|dim|bold|blink){{space_or_eol}}
+      scope: support.constant.color-attribute.gitconfig
+
+  color-named:
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto){{space_or_eol}}
+      scope: support.constant.color.gitconfig
+
+  color-numeric:
+    - match: '\b(?:{{zero_to_255}}){{space_or_eol}}'
+      scope: constant.other.color.rgb-value.gitconfig
+
+  comments:
+    - match: '[#;]'
+      scope: punctuation.definition.comment.gitconfig
+      push:
+        - meta_scope: comment.line.gitconfig
+        - match: $\n?  # consume newline so completions aren't shown at EOL when
+          pop: true    # typing comments
+
+  # The only valid escapes: '\b', '\n', '\t', '\"', '\\'.
+  escape:
+    - include: trailing-slash
+    - match: \\[^{{escape_char}}]
+      scope: invalid.illegal.escape.gitconfig
+    - match: \\[{{escape_char}}]
+      scope: constant.character.escape.gitconfig
+
+  illegal-newline:
+    - match: \n
+      scope: invalid.illegal.gitconfig
+
+  illegal-slash:
+    - match: \\
+      scope: invalid.illegal.gitconfig
+
+  # changed = red
+  # untracked = bold green
+  key-color-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: variable.other.readwrite.gitconfig
+        2: keyword.operator.assignment.gitconfig
+      push:
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
+        - match: (?=^\s*\[)                        # start of new section
+          pop: true
+        - include: color
+
+  # key = val
+  key-value-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: variable.other.readwrite.gitconfig
+        2: keyword.operator.assignment.gitconfig
+      push:
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
+        - match: (?=^\s*\[)                        # start of new section
+          pop: true
+        - include: value
+
+  # [section "subsection"]
+  section-header:
+    - match: \[
+      scope: punctuation.definition.section.begin.gitconfig
+      push:
+        - meta_scope: meta.brackets.gitconfig
```

Comment:
Maybe someone could help clarify the `definition.section` stuff here? 

1. In an INI file a `section` is the whole block under the `[section_name]` construct.
2. Per convention (PackageDev completions) I can use `punctuation.definition. ...` or `punctuation.section. ...` but not `punctuation.definition.section`.
3. [ST3 documentation / Scope Naming](https://www.sublimetext.com/docs/3/scope_naming.html) says, if you use `meta.brackets` you need to / should use `punctuation.section.brackets. (begin | end)`

All of that is quite mixed up here.

I'd suggest to handle a INI file section the same way as JSON object (`meta.mapping`) in general.
The question then is how to correctly scope its "headline"?

Reply:
I changed the scope from `punctuation.definition.section` to `punctuation.definition.brackets` to follow [the Scope Naming docs](http://www.sublimetext.com/docs/3/scope_naming.html#meta) more closely. I also have the `section` context matching the `[header]` and the whole block underneath it.